### PR TITLE
build: merge gen-proto scripts into one

### PR DESCRIPTION
**Pull request by gozssky**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflow/issues/6494

### What is changed and how it works?

Merge dm/engine gen-proto scripts into scripts/generate-protobuf.sh.
This PR also adds missing imports to `tool/check/tools.go`.

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
None
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -301,8 +302,8 @@ dm-chaos-case:
 dm_debug-tools:
 	$(GOBUILD) -ldflags '$(LDFLAGS)' -o bin/binlog-event-blackhole ./dm/debug-tools/binlog-event-blackhole
 
-dm_generate_proto: tools/bin/protoc-gen-gogofaster tools/bin/protoc-gen-grpc-gateway
-	./dm/generate-dm.sh
+dm_generate_proto:
```
Comment (lonng) [inline Makefile:305]: Can we remove the target directly?

Comment (gozssky) [reply Makefile]: It looks okay, removed.

**Review by amyangfei — APPROVED** [review]

**Review by lance6716 — APPROVED** [review]

**Comment by gozssky** [comment]
The leak-test failed in `github.com/pingcap/tiflow/cdc/kv`. https://ci2.pingcap.net/blue/organizations/jenkins/cdc_ghpr_leak_test/detail/cdc_ghpr_leak_test/32195/pipeline

**Pull request merged**